src/saga/adaptors/shell/shell_file.py

```diff
@@ -803,6 +803,31 @@ def get_size (self, tgt_in) :
     # ----------------------------------------------------------------
     #
     @SYNC_CALL
+    def exists_self (self):
+
+        self._is_valid ()
+
+        return self.exists (self.url)
+   
+   
+    # ----------------------------------------------------------------
+    #
+    @SYNC_CALL
+    def exists (self, tgt_in):
+
+        self._is_valid ()
+
+        cwdurl = saga.Url (self.url) # deep copy
+        tgt    = saga.Url (tgt_in)   # deep copy
+
+        ret, out, _ = self._command (" test -e '%s'" % tgt.path)
```

Comment:
How would this work for sftp and friends, they use the same code path, dont they?